ports/geckolib/glue.rs

```diff
@@ -3706,5 +3706,8 @@ pub extern "C" fn Servo_ProcessInvalidations(set: RawServoStyleSetBorrowed,
 
     if let Some(ref mut data) = data {
         data.invalidate_style_if_needed(element, &shared_style_context, None);
+        if data.restyle.hint.contains_subtree() {
```

Comment:
Nope, this is not enough.

You need to do something like:

```rust
if !data.restyle.hint.is_empty() {
    // ...
}
```

But that wouldn't still be enough for sibling invalidations, so you'd need to iterate over the siblings too. Also, it wouldn't be enough if we invalidated any descendant, but not ourselves.

The easiest way to do it is returning something like an `InvalidationResult` from there. Renaming `InvalidationResult` to `SingleInvalidationResult`, and using something like:

```rust
struct InvalidationResult {
    pub invalidated_self: bool,
    pub invalidated_descendants: bool,
    pub invalidated_sliblings: bool,
}
```

Or, using bitflags:

```rust
bitflags! {
    pub flags InvalidationResult: u8 {
        const INVALIDATED_SELF = 1 << 0;
        const INVALIDATED_DESCENDANTS = 1 << 1;
        const INVALIDATED_SIBLINGS = 1 << 2;
    }
}
```

Happy to help out or write it if needed.

In the case siblings were invalidated, we need to do the same we do in Gecko when getting restyle hints, that is, calling it on the parent element, and marking it as dirty.